src/order-sortedness.c

```diff
@@ -127,25 +147,45 @@ enum vctrs_sortedness dbl_sortedness(const double* p_x,
   return VCTRS_SORTEDNESS_sorted;
 }
 
+static inline int dbl_cmp_numbers(double x, double y, const int direction);
+
 /*
- * Compare two doubles, handling `na_order` and `direction`
+ * Compare two doubles, handling `na_order`, `direction`, and `na_nan_cmp`
  */
 static inline
-int dbl_cmp(double x, double y, const int direction, const int na_order) {
-  if (isnan(x)) {
-    if (isnan(y)) {
-      return 0;
-    } else {
-      return na_order;
+int dbl_cmp(const double x,
```

Comment:
Since we are not passing addresses, I would have made the signature consistent by removing all `const` qualifiers rather than add some.